lib/rust/prelude/src/data/non_empty_vec.rs

```diff
@@ -262,11 +241,53 @@ impl<T> NonEmptyVec<T> {
     ///
     /// ```
     /// use enso_prelude::NonEmptyVec;
-    /// let mut vec = NonEmptyVec::new(0, vec![1, 2]);
+    /// let mut vec = NonEmptyVec::<_, usize>::new(0, vec![1, 2]);
     /// assert_eq!(*vec.last_mut(), 2)
     /// ```
     pub fn last_mut(&mut self) -> &mut T {
-        self.get_mut(self.len() - 1).unwrap_or_else(|| unreachable!())
+        self.elems.last_mut().unwrap_or_else(|| unreachable!())
+    }
+
+    /// Convert this non-empty vector to vector.
+    pub fn into_vec(self) -> Vec<T> {
+        self.elems.into()
+    }
+
+    /// Consume this non-empty vector, map each element with a function, and produce a new one.
+    pub fn mapped<S>(self, f: impl FnMut(T) -> S) -> NonEmptyVec<S> {
+        let elems = self.elems.into_iter().map(f).collect();
+        NonEmptyVec { elems }
+    }
+}
+
+impl<T, I> NonEmptyVec<T, I>
+where I: vec_indexed_by::Index
+{
+    /// Obtain a mutable reference to teh element in the vector at the specified `index`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use enso_prelude::NonEmptyVec;
+    /// let mut vec = NonEmptyVec::new(0, vec![1, 2]);
+    /// let reference = vec.get_mut(0);
+    /// assert!(reference.is_some());
+    /// assert_eq!(*reference.unwrap(), 0);
+    /// ```
+    pub fn get_mut(&mut self, index: I) -> Option<&mut T> {
+        self.elems.get_mut(index)
+    }
+
+    /// Get the tail reference.
+    pub fn tail(&mut self) -> &[T]
+    where I: From<usize> {
+        &self.elems[I::from(1_usize)..]
+    }
+
+    /// Get the mutable tail reference.
+    pub fn tail_mut(&mut self) -> &mut [T]
+    where I: From<usize> {
+        &mut self.elems[I::from(1_usize)..]
```

Comment:
The only standard integer type that implements `From<usize>` is `usize`. A `From<u8>` bound would allow this API to be used for any positive integer index type.

Reply:
Oh, good point!